### ISSUE TITLE
X-Ray exporter: `:*` suffix in `aws.cloudwatch_logs.arn` trips up the CloudWatch portal

### ISSUE THREAD

**Issue by mderriey** [aws-observability/aws-otel-collector#1865]
**Describe the question**
I'm wondering whether the X-Ray exporter should remove the potential `:*` suffix in the log group ARN as it prevents the CloudWatch portal from getting the log records associated with a trace.

**Steps to reproduce if your question is related to an action**
We use the `aws-otel-collector` to export traces to X-Ray.

We want to get traces-to-logs correlation in the CloudWatch portal, so we enrich traces with log group information.

We achieve this through the `OTEL_RESOURCE_ATTRIBUTES` environment variable, which value looks like `aws.log.group.arns=arn:aws:logs:<region>:<account-id>:log-group:<log-group-name>:*`.

Please note the trailing `:*` segment.
We're setting this via CDK, and we're not transforming the log group ARN that CDK gives us.
When we navigate to the log group in the CloudWatch portal, the displayed ARN also shows the `:*` suffix.

**What did you expect to see?**

When inspecting a specific trace through the CloudWatch portal, we expect to see log records associated with this trace in the bottom section, but it's empty.

![image](https://user-images.githubusercontent.com/6102639/220701415-815722dc-2425-46a1-ad52-a5ac5da33c8b.png)

What we can see in the Network tab of DevTools is a request that comes back with a 400 response.

URL: `https://logs.<region>.amazonaws.com/`

Request payload:

```json
{
  "query": "SOURCE \"arn:aws:logs:<region>:<account-id>:log-group:<log-group-name>:*\" START=2023-02-22T14:52:54.982Z END=2023-02-22T14:53:40.167Z | SOURCE \"<log-group-name>\" START=2023-02-22T14:52:54.982Z END=2023-02-22T14:53:40.167Z | fields @log, @timestamp, @message\n| filter @message like \"1-63f62c55-272d0f1b4530774a41bb2e47\"\n| sort @timestamp, @message desc",
  "endTime": 1677077620,
  "logGroupNames": [
    "arn:aws:logs:<region>:<account-id>:log-group:<log-group-name>:*",
    "<log-group-name>"
  ],
  "queryString": "fields @log, @timestamp, @message\n| filter @message like \"1-63f62c55-272d0f1b4530774a41bb2e47\"\n| sort @timestamp, @message desc",
  "startTime": 1677077574
}
```

Response payload:

```json
{
  "__type":"InvalidParameterException",
  "message":"1 validation error detected: Value '[arn:aws:logs:<region>:<account-id>:log-group:<log-group-name>:*, <log-group-name>]' at 'logGroupNames' failed to satisfy constraint: Member must satisfy constraint: [Member must have length less than or equal to 2048, Member must have length greater than or equal to 1, Member must satisfy regular expression pattern: [\\w+=/:,.@-]*]"}
```

The API doesn't "like" the `*` in the log group ARN.

**Environment**
Describe any aspect of your environment.
If this is related to a deployment of the ADOT Collector please
provide your Collector config file. 

**Additional context**
Sort of related to #1766

**Comment by mderriey** [comment]
Quick update, we got this working by removing the `:*` suffix in the log group ARN when we're setting the `OTEL_RESOURCE_ATTRIBUTES` environment variable.

It still feels like this should work out of the box.

Happy to provide more details where needed.

**Comment by willarmiros** [comment]
Hi @mderriey - thanks for raising this. I have cut an issue to our console team so they can support both ARN formats as both are valid. Is it safe to say that including the trailing `:*` is the reason for the 400 Bad Request errors you mentioned in #1766?

**Comment by mderriey** [comment]
I think it is, although they are different requests that I observed in that other issue, and there was no response payload, so I can't say for sure.

I got it working after removing the `:*` suffix, so there's a high chance it's what caused the issue.

Thanks a lot for getting back to me, and for passing this on to the appropriate team, let me know how that goes, happy to do some testing by including it back when it should be supported.

**Comment by willarmiros** [comment]
Fair enough. We will update likely the other issue once there is a fix in place.

**Comment by willarmiros** [comment]
The CloudWatch Console team has fixed this issue on our end, so the correlation should work with or without the `:*` suffix! Please let us know if there are still issues, but I'll close this for now!

**Comment by mderriey** [comment]
Hey 👋

Apologies for not getting back to you quicker.
I just had a chance to test it without the trailing `:*` in the log group ARN and it worked a treat.

Thanks for making this happen so quickly, it's much appreciated 🙏
